packages/demo/src/configuration/blocks/effects/lutBlock.ts

```diff
@@ -0,0 +1,176 @@
+                    mediump vec2 lowerBoundBlueSlice;
+                    lowerBoundBlueSlice.y = floor(floor(unboundBlueSlice) / blueSlicesIn2DLayout.x);
+                    lowerBoundBlueSlice.x = floor(unboundBlueSlice) - (lowerBoundBlueSlice.y * blueSlicesIn2DLayout.x);
+
+                    // Find the nearest upper bound quad location
+                    mediump vec2 upperBoundBlueSlice;
+                    upperBoundBlueSlice.y = floor(ceil(unboundBlueSlice) / blueSlicesIn2DLayout.x);
+                    upperBoundBlueSlice.x = ceil(unboundBlueSlice) - (upperBoundBlueSlice.y * blueSlicesIn2DLayout.x);
+
+                    // Within the slices, find the position of the rg values in UV space
+                    highp vec2 texPos1 = (lowerBoundBlueSlice / blueSlicesIn2DLayout) + vec2(0.5, 0.5) / texSize + (((vec2(1., 1.) / blueSlicesIn2DLayout) - vec2(1., 1.) / texSize) * color.rg);
+                    highp vec2 texPos2 = (upperBoundBlueSlice / blueSlicesIn2DLayout) + vec2(0.5, 0.5) / texSize + (((vec2(1., 1.) / blueSlicesIn2DLayout) - vec2(1., 1.) / texSize) * color.rg);
+
+                    // Get the two colors that we will interpolate in the LUT texture
+                    lowp vec4 newColor1 = texture2D(lookupTable, texPos1);
+                    lowp vec4 newColor2 = texture2D(lookupTable, texPos2);
+
+                    // Interpolate between the two depending on our blue channel
+                    lowp vec4 newColor = mix(newColor1, newColor2, fract(unboundBlueSlice));
+                    return newColor;
+                }
+            `,
+            },
+        ],
+    },
+});
+
+/**
+ * The shader bindings for the Contrast block.
+ */
+export class LUTShaderBinding extends ShaderBinding {
+    private readonly _inputTexture: RuntimeData<ConnectionPointType.Texture>;
+    private readonly _lutTexture: RuntimeData<ConnectionPointType.Texture>;
+    private readonly _level: RuntimeData<ConnectionPointType.Float>;
+
+    /**
+     * Creates a new shader binding instance for the Contrast block.
+     * @param parentBlock - The parent block
+     * @param inputTexture - The input texture
+     * @param lutTexture - The LUT input texture
+     * @param level - The level of the effect
+     */
+    constructor(
+        parentBlock: IDisableableBlock,
+        inputTexture: RuntimeData<ConnectionPointType.Texture>,
+        lutTexture: RuntimeData<ConnectionPointType.Texture>,
+        level: RuntimeData<ConnectionPointType.Float>
+    ) {
+        super(parentBlock);
+        this._inputTexture = inputTexture;
+        this._lutTexture = lutTexture;
+        this._level = level;
+    }
+
+    getBlueSlicesIn2DLayout(): Vector2 {
+        return new Vector2(2.0 /**numBlueSlicesPerRowIn2DLayout */, 17.0 /**numBlueSlicesPerColumnIn2DLayout */);
+    }
+
+    calculateTextureSize(): Vector2 {
+        return this.getBlueSlicesIn2DLayout().multiply({
+            x: 17.0,
```

Comment:
why 17? Is this a constant we can define somewhere?

Reply:
Yup, added config!